homeassistant/components/mqtt/notify.py

```diff
@@ -0,0 +1,393 @@
+"""Support for MQTT notify."""
+from __future__ import annotations
+
+import functools
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components import notify
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE, CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import EVENT_DEVICE_REGISTRY_UPDATED
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.util import slugify
+
+from . import PLATFORMS, MqttCommandTemplate
+from .. import mqtt
+from .const import (
+    ATTR_DISCOVERY_HASH,
+    ATTR_DISCOVERY_PAYLOAD,
+    CONF_COMMAND_TEMPLATE,
+    CONF_COMMAND_TOPIC,
+    CONF_ENCODING,
+    CONF_QOS,
+    CONF_RETAIN,
+    DOMAIN,
+)
+from .discovery import MQTT_DISCOVERY_DONE, MQTT_DISCOVERY_UPDATED, clear_discovery_hash
+from .mixins import (
+    CONF_CONNECTIONS,
+    CONF_IDENTIFIERS,
+    MQTT_ENTITY_DEVICE_INFO_SCHEMA,
+    async_setup_entry_helper,
+    cleanup_device_registry,
+    device_info_from_config,
+)
+
+CONF_TARGETS = "targets"
+CONF_TITLE = "title"
+
+MQTT_NOTIFY_SERVICES_SETUP = "mqtt_notify_services_setup"
+
+PLATFORM_SCHEMA = mqtt.MQTT_BASE_PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_COMMAND_TOPIC): mqtt.valid_publish_topic,
+        vol.Optional(CONF_COMMAND_TEMPLATE): cv.template,
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_TARGETS, default=[]): cv.ensure_list,
+        vol.Optional(CONF_TITLE, default=notify.ATTR_TITLE_DEFAULT): cv.string,
+        vol.Optional(CONF_RETAIN, default=mqtt.DEFAULT_RETAIN): cv.boolean,
+    }
+)
+
+DISCOVERY_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_DEVICE): MQTT_ENTITY_DEVICE_INFO_SCHEMA,
+    },
+    extra=vol.REMOVE_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up MQTT climate device dynamically through MQTT discovery."""
+    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+    setup = functools.partial(_async_setup_notify, hass, config_entry=config_entry)
+    await async_setup_entry_helper(hass, notify.DOMAIN, setup, DISCOVERY_SCHEMA)
+
+
+async def _async_setup_notify(
+    hass,
+    legacy_config: ConfigType,
+    config_entry: ConfigEntry,
+    discovery_data: dict[str, Any],
+):
+    """Set up the MQTT notify service with auto discovery."""
+    config = DISCOVERY_SCHEMA(discovery_data[ATTR_DISCOVERY_PAYLOAD])
+    service_name = slugify(config.get(CONF_NAME) or DOMAIN)
```

Comment:
Not sure there is an `object_id` since there is no entity created and ther is no `State`. Is requiring a name really needed? If there targets specified, the sevice name's will become `notity.mqtt_{target_name}`.

Reply:
made `name` config parameter required, still need to update docs.

Reply:
docs PR was updated